module/input_handler.py

```diff
@@ -0,0 +1,133 @@
+from module.command_node import NodeType
+from module.frame_functions import cast_and_send_ui_frame
+
+
+class input_handler:
+    def __init__(self, cli_controller):
+        self.cli_controller = cli_controller
+
+    @staticmethod
+    def print_command_not_found(command: str) -> None:
+        """
+        Prints command not found text with the command parameter.
+        """
+        print("\tCommand '{}' not found, type 'help' for possible commands.".format(command))
+
+    @staticmethod
+    def check_amount_parameters(parameters: list, required_amount: int):
+        """
+        compares the length of the parameters list to the required amount,
+        returns true when the length of the parameters is equal to the required amount
+        prints to user if not enough or too many parameters
+        """
+        if len(parameters) < required_amount or len(parameters) > required_amount:
+            print("\tExpected {} parameters, got {}.".format(required_amount, len(parameters)))
+            return False
+        return True
+
+    def handle_help(self, help_parameters):
+        try:
+            node = self.cli_controller.current_node
+
+            for param in help_parameters:
+                node = node[param.upper()]
+
+            self.cli_controller.print_help(node)
+        except IndexError:
+            self.cli_controller.print_help(self.cli_controller.current_node)
+        except KeyError:
+            self.print_command_not_found(param)
```

Comment:
Are you sure param can be defined at this pound in the code since it is assigned in the for loop. So can the param still exist outside of the for loop.

Reply:
Yes, as this function can only be called if help_parameters is at least one, so param will be assigned at least once.